src/System.Data.SqlClient/src/Interop/SNINativeMethodWrapper.Windows.cs

```diff
@@ -375,7 +375,7 @@ internal static unsafe uint SNISecGenClientContext(SNIHandle pConnectionObject,
                     ref sendLength,
                     out local_fDone,
                     pin_serverUserName,
-                    (uint)(serverUserName == null ? 0 : serverUserName.Length),
```

Comment:
serverUserName was already dereferenced above, so if it's null, the above fixed statement will null ref